### PR TITLE
18887 - Create Schema

### PR DESCRIPTION
**Pull request by hfekete**
*Issue:18887 *https://github.com/bcgov/entity/issues/18887 

*Description of changes:*
- adding JSON schema to verify API requests
- some updates to the backend
- started adding tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

File: btr-api/src/btr_api/schemas/SignificantIndividualsFiling.json [hunk 1]
```diff
@@ -0,0 +1,176 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "definitions": {
+    "FilingActionE": {
+      "enum": ["add", "edit", "remove"],
```
Comment (thorwolpert) [inline btr-api/src/btr_api/schemas/SignificantIndividualsFiling.json:5]: i don't think we should have actions like this.
in other filings, and it looks like bods is the same, that a submission is for the entire current state.
as such, if a person has an identifier they get updated, if they don't have one they're created and if they are in the longitudinal record but not in the filing they are removed.

File: btr-api/src/btr_api/schemas/SignificantIndividualsFiling.json [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+          "type": "boolean"
+        },
+        "inConcertControl": {
+          "type": "boolean"
+        }
+      },
+      "type": "object",
+      "required": ["registeredOwner", "beneficialOwner", "indirectControl", "inConcertControl"]
+    },
+    "CitizenshipTypeE": {
```
Comment (thorwolpert) [inline btr-api/src/btr_api/schemas/SignificantIndividualsFiling.json:44]: In all our schemas, and in openownership the elements are all camelCase.
Please follow that standard

File: btr-api/src/btr_api/resources/json_schema.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+from flask import Blueprint
+from flask import jsonify
+
+from btr_api.services.schema import SchemaService
+
+bp = Blueprint("json-schemas", __name__)
+
+
+@bp.route("/", methods=("GET",))
+def list_all_schemas():
```
Comment (thorwolpert) [inline btr-api/src/btr_api/resources/json_schema.py:42]: We would probably statically publish these, so it's not needed to add an active end-point for this.

**Review by thorwolpert — CHANGES_REQUESTED** [review]
the schema naming should follow our existing standards, which are also inline with bods

**Comment by kialj876** [comment]
Looks good to me, but can you update it to run the tests successfully? Looks like it errs to run due to an import issue on the test_schema...

**Review by kialj876 — APPROVED** [review]

**Pull request merged**